Lua14/Lua/Userdata/ComponentIndexUserdata.cs

```diff
@@ -0,0 +1,38 @@
+﻿using NLua;
+using Robust.Shared.GameObjects;
+
+namespace Lua14.Lua.Userdata;
+
+public sealed class ComponentIndexUserdata : LuaUserdata
+{
+    [LuaMember(Name = "type")]
+    public Type Type;
+
+    public ComponentIndexUserdata(NLua.Lua lua, Type type) : base(lua)
+    {
+        if (!typeof(IComponent).IsAssignableFrom(type))
+            throw new Exception("Cant create a ComponentIndex because the specified type is not a component.");
+
+        Type = type;
+    }
+
+    protected override object? Call(LuaUserdata self, params object[] args)
+    {
+        if (args.Length < 1)
+            return Activator.CreateInstance(Type);
+
+        if (args[0] is not LuaTable parameters)
+            throw new Exception("First argument should be a table.");
+
+        var paramsDict = Lua.TableToDictionary<string, object?>(parameters) ?? throw new Exception("First argument should be a table<string, any>.");
+        var comp = Activator.CreateInstance(Type)!;
```

Comment:
Adding an InvalidOperationException instead of asserting could be useful.
```suggestion
       var comp = Activator.CreateInstance(Type) 
               ?? throw new InvalidOperationException("Could not create instance of type.");
```